src/map_io/map_object_saver.cc

```diff
@@ -98,27 +95,27 @@ Serial MapObjectSaver::register_object(const MapObject& obj) {
 
 	assert(!rec.registered);
 
-	if (dynamic_cast<Flag const*>(&obj)) {
+	if (obj.descr().type() == MapObjectType::FLAG) {
```

Comment:
I think we could gain even more performance by sticking the `==` comparisons into a switch statement and then handling the `>=` within its default case.